opennms-config-model/src/main/java/org/opennms/netmgt/config/vacuumd/Action.java

```diff
@@ -96,6 +96,10 @@ public void setStatement(final Statement statement) {
         m_statement = ConfigUtils.assertNotNull(statement, "statement");
     }
 
+    public boolean hasStatement(){
+        return m_statement != null;
```

Comment:
Is `null` a legal state for `m_statement`? The XML annotation implies it is required. Under what circumstances is it actually legal for it to be null?

If it is illegal for it to be null we could update the getter to assert non-null with an `IllegalStateException` or ideally post-deserialization do that to guarantee that invariant is checked.

The fact that an NPE is blowing up deeper in the stack might just be an indication that we created an illegal object earlier on.

---

If it is in fact legal to be `null` we can explicitly indicate that to our callers by refactoring the getter to return an Optional. We can then get rid of this `hasStatement` method.

Reply:
Forgot to mention, the setter also implies that null is illegal. So pretty sure this shouldn't be allowed null.

Reply:
Thanks Matthew. That's a very good point. The statement could not be null, it would throw exception by xml read before reaching the line of the code.

Reply:
Ok, if we are confident that `null` is an illegal state for this type and that we already guard against that invariant let's remove the new redundant code that checks that.